camera/android/src/main/java/com/capacitorjs/plugins/camera/CameraPlugin.java

```diff
@@ -454,20 +453,33 @@ private void returnBase64(PluginCall call, ExifWrapper exif, ByteArrayOutputStre
         call.resolve(data);
     }
 
-    @Nullable
     @Override
-    protected String overrideStateForPermission(Permission permission) {
-        if (permission.alias().equals("camera")) {
-            // If the camera permission is defined in the manifest, then we have to prompt the user
-            // or else we will get a security exception when trying to present the camera. If, however,
-            // it is not defined in the manifest then we don't need to prompt and it will just work.
-            if (hasDefinedPermissions(new Permission[] { permission })) {
-                return null;
-            } else {
-                return "granted";
-            }
+    @PluginMethod
+    public void requestPermissions(PluginCall call) {
+        // If the camera permission is defined in the manifest, then we have to prompt the user
+        // or else we will get a security exception when trying to present the camera. If, however,
+        // it is not defined in the manifest then we don't need to prompt and it will just work.
+        if (hasDefinedPermissions(new String[] { Manifest.permission.CAMERA })) {
+            // Request all permissions (includes camera)
+            super.requestPermissions(call);
+        } else {
```

Comment:
This did not handle the situation where the app only requested 'camera' permissions. It should be superseded now by the latest changes.